minions/minion.py

```diff
@@ -63,13 +69,19 @@ def __init__(
         self.remote_client = remote_client
         self.max_rounds = max_rounds
         self.callback = callback
+        self.log_dir = log_dir
+
+        # Create log directory if it doesn't exist
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
 
     def __call__(
         self,
         task: str,
         context: List[str],
         max_rounds=None,
         doc_metadata=None,
+        task_id=None,
```

Comment:
Can you explain what `task_id` means here? it's not the `task_id` that we use inside the commands in minions, right? so maybe change to a different name to avoid confusion?